Broken CI for conda on windows

Hi,

https://github.com/stack-of-tasks/eigenpy/runs/7976829564?check_suite_focus=true shows:
```
[  5%] Building CXX object CMakeFiles/eigenpy.dir/src/solvers/preconditioners.cpp.obj
	C:\Miniconda\envs\eigenpy\Library\bin\cmake.exe -E cmake_cl_compile_depends --dep-file=CMakeFiles\eigenpy.dir\src\solvers\preconditioners.cpp.obj.d --working-dir=D:\a\eigenpy\eigenpy\build --filter-prefix="Note: including file: " -- C:\PROGRA~1\MICROS~2\2022\ENTERP~1\VC\Tools\MSVC\1433~1.316\bin\Hostx64\x64\cl.exe @C:\Users\RUNNER~1\AppData\Local\Temp\nm745A.tmp
preconditioners.cpp
C:\Miniconda\envs\eigenpy\Library\include\boost/assert/source_location.hpp(97): error C2039: '_snprintf': is not a member of 'std'
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.33.31629\include\list(24): note: see declaration of 'std'
C:\Miniconda\envs\eigenpy\Library\include\boost/assert/source_location.hpp(104): error C2039: '_snprintf': is not a member of 'std'
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.33.31629\include\list(24): note: see declaration of 'std'
NMAKE : fatal error U1077: 'C:\Miniconda\envs\eigenpy\Library\bin\cmake.exe' : return code '0x2'
```

Looking at boost/assert/source_location.hpp:97, this is in `source_location::to_string` function, which was added in boost 1.78:
https://www.boost.org/doc/libs/1_80_0/libs/assert/doc/html/assert.html#changes_in_1_78_0

boost/convert/detail/config.hpp:55 is : `#   define snprintf _snprintf`, but it shouldn't be active as this is only for MSVC < 14, and we are on 19.

But this could come from cpython. We are on 3.7.12 in that CI test, and https://github.com/python/cpython/blob/v3.7.12/Include/pyerrors.h#L491 is also `# define snprintf _snprintf`

I'll try to define `HAVE_SNPRINTF` for windows to see if this fix the CI.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -233,6 +233,7 @@ if(NOT WIN32)
 else()
   target_compile_options(${PROJECT_NAME}
                          PRIVATE $<$<CXX_COMPILER_ID:MSVC>:-bigobj>)
+  target_compile_definitions(${PROJECT_NAME} PUBLIC "HAVE_SNPRINTF")
 endif()
 
 target_link_boost_python(${PROJECT_NAME} PUBLIC)

